src/printer-python.js

```diff
@@ -0,0 +1,179 @@
+"use strict";
+
+const docBuilders = require("./doc-builders");
+const concat = docBuilders.concat;
+const join = docBuilders.join;
+const hardline = docBuilders.hardline;
+const line = docBuilders.line;
+const softline = docBuilders.softline;
+const group = docBuilders.group;
+const indent = docBuilders.indent;
+const ifBreak = docBuilders.ifBreak;
+
+function printArguments(print, path, argsKey, defaultsKey) {
+  const n = path.getValue();
+
+  // python AST represent arguments and default
+  // value in two different lists, so we grab
+  // the list of the arguments and the list of
+  // default values and we merge them together and sort
+  // them by column. Then we iterated one by one and
+  // if the next element is a default value we merge it with
+  // the current one
+
+  const merge = [...n[argsKey], ...n[defaultsKey]].sort(
+    (a, b) => a.col_offset - b.col_offset
+  );
+
+  const parts = [];
+
+  let currentArgument = 0;
+  let currentDefault = 0;
+
+  for (let i = 0; i < merge.length; i++) {
+    const next = merge[i + 1];
+
+    const part = [path.call(print, argsKey, currentArgument)];
+
+    currentArgument += 1;
+
+    if (next && next.ast_type != "arg") {
+      part.push("=", path.call(print, defaultsKey, currentDefault));
+
+      i += 1;
+      currentDefault += 1;
+    }
+
+    parts.push(concat(part));
+  }
+
+  return parts;
+}
+
+function genericPrint(path, options, print) {
+  const n = path.getValue();
+  if (!n) {
+    return "";
+  }
+
+  if (typeof n === "string") {
+    return n;
+  }
+
+  switch (n.ast_type) {
+    case "Module": {
+      return concat([
+        join(concat([hardline, hardline]), path.map(print, "body")),
+        hardline
+      ]);
+    }
+
+    case "FunctionDef": {
+      return concat([
+        "def ",
+        path.call(print, "name"),
+        group(
+          concat([
+            "(",
+            indent(concat([softline, path.call(print, "args")])),
+            softline,
+            ")"
+          ])
+        ),
+        ":",
+        indent(concat([line, concat(path.map(print, "body"))]))
+      ]);
+    }
+
+    case "arguments": {
+      let parts = printArguments(print, path, "args", "defaults");
+
+      // add varargs (*args)
+
+      if (n.vararg) {
+        parts.push(concat(["*", path.call(print, "vararg")]));
+      }
+
+      // add keyword arguments (**kwargs)
+
+      if (n.kwarg) {
+        parts.push(concat(["**", path.call(print, "kwarg")]));
+      }
+
+      if (n.kwonlyargs.length > 0) {
```

Comment:
kwonlyargs need to go before `**kwargs`

Reply:
@taion why? This seems to work just fine :)

<img width="525" alt="screen shot 2017-11-29 at 21 17 45" src="https://user-images.githubusercontent.com/667029/33399506-c7c6cd4a-d54a-11e7-9fb0-6ffa28c0d3e1.png">

Reply:
![image](https://user-images.githubusercontent.com/3112159/33400130-f931fe18-d522-11e7-8a8d-5b8e59fe9609.png)

Reply:
oh, got it! thanks!